### PR TITLE
treewide: replace `lib/${python.libPrefix}/site-packages` with its sh…

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…orthand `${python.sitePackages}`

which should be always preferred

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
rebuilds are triggered because changes like the following:

```diff
-eapth="$out/lib/${python.libPrefix}"/site-packages/easy-install.pth
+eapth="$out/${python.sitePackages}/easy-install.pth"
```

**Comment by pbsds** [comment]
from ofborg:
```
 error: attribute 'sitePackage' missing

       at /ofborg/checkout/3/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-1/pkgs/development/python-modules/selenium/default.nix:37:23:

           36|   postInstall = ''
           37|     DST_PREFIX=$out/${python.sitePackage}/selenium/webdriver/
             |                       ^
           38|     DST_REMOTE=$DST_PREFIX/remote/
       Did you mean sitePackages?
```

File: doc/hooks/python.section.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Python {#setup-hook-python}
 
-Adds the `lib/${python.libPrefix}/site-packages` subdirectory of each build input to the `PYTHONPATH` environment variable.
+Adds the `${python.sitePackages}` subdirectory of each build input to the `PYTHONPATH` environment variable.
```
Comment (pbsds) [inline doc/hooks/python.section.md:3]:
```suggestion
Adds the `python.sitePackages` subdirectory (i.e. `lib/pythonX.Y/site-packages`) of each build input to the `PYTHONPATH` environment variable.
```

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 222817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>gramps</li>
    <li>gramps.dist</li>
  </ul>
</details>

**Pull request merged**